network/src/protocols/network/mod.rs

```diff
@@ -311,34 +337,117 @@ impl<TMessage> FusedStream for NetworkEvents<TMessage> {
 /// interface they need.
 ///
 /// Provide Protobuf wrapper over `[peer_manager::PeerManagerRequestSender]`
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub struct NetworkSender<TMessage> {
     peer_mgr_reqs_tx: PeerManagerRequestSender,
     connection_reqs_tx: ConnectionRequestSender,
-    _marker: PhantomData<TMessage>,
+    serializing_message_sender: aptos_channel::Sender<(), MessageSerializationRequest<TMessage>>,
```

Comment:
I'm pretty sure that adding another queue in the sending path isn't what we want. I'd vote more strongly against this now if I weren't planning on replacing it in a couple months.

Reply:
Hmm... I'm not sure there's a way of parallelizing the sends without introducing another queue (without a huge code refactor). I'm gonna close this for now and we can reopen it if we see the need (e.g., for performance).